### PR TITLE
Encrypt outData in work order requests in listener mode

### PR DESCRIPTION
**Pull request by manju956**
- Add encryption of outData if present in work order request in signature.py
- Fixes request decryption failure issue if work order request contains outData

Signed-off-by: manju956 <manjunath.a.c@intel.com>

### CODE REVIEW

File: common/crypto_utils/avalon_crypto_utils/signature.py [hunk 1]
```diff
@@ -85,8 +85,8 @@ def __payload_json_check(self, json_data):
         return param_valid
 
 # -----------------------------------------------------------------------------
-    def __encrypt_workorder_indata(
-            self, input_json_params, session_key, session_iv,
+    def __encrypt_workorder_data(
+            self, input_json_params, data_type, session_key, session_iv,
```
Comment (manojgop) [inline common/crypto_utils/avalon_crypto_utils/signature.py:89]: Insted of passing data type pass the in/out data array to be encrypted. Please refer 
https://github.com/hyperledger/avalon/blob/master/examples/graphene_apps/python_worker/avalon_worker/work_order_processor.py#L339
and https://github.com/hyperledger/avalon/blob/master/examples/graphene_apps/python_worker/avalon_worker/crypto/worker_encryption.py#L292 and

Comment (manju956) [reply common/crypto_utils/avalon_crypto_utils/signature.py]: Done

File: common/crypto_utils/avalon_crypto_utils/signature.py [hunk 2]
```diff
@@ -248,9 +247,21 @@ def generate_client_signature(
...
-        self.__encrypt_workorder_indata(
-            input_json_params, session_key,
-            session_iv, worker.encryption_key, data_key, data_iv)
+
+        logger.debug("Encrypting Workorder inData")
+        in_data_objects = input_json_params['inData']
+        updated_in_data_objects = self.__encrypt_workorder_data(
+            in_data_objects, session_key, session_iv,
+            worker.encryption_key, data_key, data_iv)
+        input_json_params['inData'] = updated_in_data_objects
```
Comment (manojgop) [inline common/crypto_utils/avalon_crypto_utils/signature.py:256]: I think `in_data_objects` will be modified in-place by `__encrypt_workorder_data()`. So you don't need to return `updated_in_data_objects ` and reassign it

Comment (manju956) [reply common/crypto_utils/avalon_crypto_utils/signature.py]: Removed not needed re-assignment.

File: common/crypto_utils/avalon_crypto_utils/signature.py [hunk 3]
```diff
@@ -248,9 +245,19 @@ def generate_client_signature(
 
         encrypted_session_key_str = byte_array_to_hex_str(
             encrypted_session_key)
-        self.__encrypt_workorder_indata(
-            input_json_params, session_key,
-            session_iv, worker.encryption_key, data_key, data_iv)
+
+        logger.debug("Encrypting Workorder inData")
```
Comment (manojgop) [inline common/crypto_utils/avalon_crypto_utils/signature.py:249]: We don't need to assume inData will be always present in work order request. We could have some workloads which does some fixed functionality without inData. You could add below check for inData similar to outData
````
 if "inData" in input_json_params and \
                len(input_json_params["inData"]) > 0:
```

Comment (manju956) [reply common/crypto_utils/avalon_crypto_utils/signature.py]: Done

**Pull request merged**